acestep/ui/gradio/events/training/lora_training.py

```
@@ -0,0 +1,304 @@
+                log_lines = log_lines[-15:]
+            log_text = "\n".join(log_lines)
+
+            if step > 0 and loss is not None and loss == loss:  # NaN check
+                step_list.append(step)
+                loss_list.append(float(loss))
+
+            plot_figure = _training_loss_figure(training_state, step_list, loss_list)
+            yield display_status, log_text, plot_figure, training_state
+
+            if training_state.get("should_stop", False):
+                logger.info("ℹ️ Training stopped by user")
```

Ruff (0.15.0)

[warning] String contains ambiguous `ℹ` (INFORMATION SOURCE). Did you mean `i` (LATIN SMALL LETTER I)?

(RUF001)